lib/kafka/producer.rb

```diff
@@ -206,14 +206,18 @@ def produce(value, key: nil, headers: {}, topic:, partition: nil, partition_key:
         create_time: create_time
       ))
 
+      if message.bytesize >= @max_buffer_bytesize
+        buffer_overflow topic, "Message is too big: message_bytesize=#{message.bytesize}"
```

Comment:
@yboulkaid where is `buffer_overflow` function defined?

Reply:
It's defined in the `AsyncProducer`: https://github.com/workato/ruby-kafka/blob/b9b1cdf0fd900b5536df3127d4bb0907c7326923/lib/kafka/async_producer.rb#L182